server/grpc/api/mapping.go

```diff
@@ -1,13 +1,18 @@
 package api
 
 import (
+	"fmt"
+
 	"github.com/mesg-foundation/core/protobuf/definition"
 	"github.com/mesg-foundation/core/service"
 )
 
 // FromProtoService converts a the protobuf definition to the internal service struct
 // TODO: should not be public. Need to move server/grpc/service.go to server/grpc/api/service.go
 func FromProtoService(s *definition.Service) (*service.Service, error) {
+	if s == nil {
```

Comment:
Do not check for nil, as it only hides some real issue with nil pointers. It should just do conversion without returning an error.

Reply:
This check should exist somewhere. I guess it could be move in the `service.Create` api function to keep the mapping simple 👍